Doctrine/UserManager.php

```diff
@@ -28,30 +28,41 @@ class UserManager extends BaseUserManager
     /**
      * @var string
      */
+    private $className;
+
+    /**
+     * @var string
+     */
     protected $class;
 
     /**
      * @var ObjectRepository
      */
-    protected $repository;
+    private $repository;
 
     /**
      * Constructor.
      *
      * @param PasswordUpdaterInterface $passwordUpdater
      * @param CanonicalFieldsUpdater   $canonicalFieldsUpdater
      * @param ObjectManager            $om
-     * @param string                   $class
+     * @param string                   $className
      */
-    public function __construct(PasswordUpdaterInterface $passwordUpdater, CanonicalFieldsUpdater $canonicalFieldsUpdater, ObjectManager $om, $class)
+    public function __construct(PasswordUpdaterInterface $passwordUpdater, CanonicalFieldsUpdater $canonicalFieldsUpdater, ObjectManager $om, $className)
     {
         parent::__construct($passwordUpdater, $canonicalFieldsUpdater);
 
         $this->objectManager = $om;
-        $this->repository = $om->getRepository($class);
+        $this->className = $className;
```

Comment:
can we instead use a single property ? then, in ``getClass``, we would check whether there is a ``:`` in the name and resolve it with Doctrine (getting it from the metadata allowed supporting Doctrine aliases like ``AppBundle:User``) and update the property (to avoid resolving it again and again).
I also suggest throwing a deprecation warning when using a Doctrine alias for the class.